src/drivers/interrupt/cortexm_nvic.c

```diff
@@ -48,8 +55,58 @@ static uint32_t exception_table[EXCEPTION_TABLE_SZ] __attribute__ ((aligned (128
 extern void *trap_table_start;
 extern void *trap_table_end;
 
-void interrupt_handle(void) {
+struct cpu_saved_ctx {
+	uint32_t r[5];
+	uint32_t lr;
+	uint32_t pc;
+	uint32_t psr;
+};
+
+struct irq_saved_state {
+	struct cpu_saved_ctx ctx;
+	uint32_t sp;
+	uint32_t lr;
+};
+
+extern void __irq_trampoline(struct irq_saved_state *state, struct context *regs);
+extern void __pending_handle(void);
+extern void __pendsv_handle(void);
+extern void interrupt_handle_enter(void);
+
+/*
+ * Usual interrupt handling:
+ *     execution flow
+ *         |
+ *         |
+ *         v
+ *         |--------> interrupt_handle (irq context)
+ *                        |
+ *         <--------------|
+ *         |
+ *         |
+ *         v
+ *
+ * This method:
```

Comment:
Yes, I think so. But may be not very representative.
In my opinion, it is really hard to keep this code is C, because of compiler-specifiec push's on function entering, cheating with sp, and some other issues. I believe that it is not very big bunch of functions in asm.